Collect asset sizes from build

The bundle sizes of some modules are collected during build in `dist/stats.json`. However, the sizes of assets are not reported.

The goal here is to augment that stats file with file sizes of the assets bundled from `assets/`.

See `src/tools/scripts/stats/index.ts` and `src/tools/scripts/build.ts`.

Hi! I'm trying to install the dependencies using `npm ci` in gitpod but getting an authentication error : 
`npm ERR! 401 Unauthorized - GET https://npm.pkg.github.com/download/@sugarlabs/musicblocks-v4-lib/0.2.0/f169ab57da6e2c473962d1e2575b3832bb18c35f - authentication token not provided`
I'm new and was just looking around good first issues. Any input on this error is appreciated :) 
thanks!

--- a/src/@types/app.d.ts
+++ b/src/@types/app.d.ts
@@ -39,7 +39,7 @@ export interface IAppConfig {
 
 /** Type definition for the map of items imported at app load. */
 export type TAppImportMap = {
-    lang: TI18nLang | undefined;
+    lang: Partial<Record<TI18nLang, boolean>>;
     assets: Record<string, boolean>;
     components: Partial<Record<TComponentId, boolean>>;
 };

--- a/src/@types/env.d.ts
+++ b/src/@types/env.d.ts
@@ -9,13 +9,3 @@ declare module '*.svg' {
 declare module '*.jsonc';
 
 declare module '*.wasm';
-
-declare module 'virtual:stats' {
-    const stats:
-        | {
-              i18n: Record<string, number>;
-              assets: Record<string, number>;
-              modules: Record<string, number>;
-          }
-        | undefined;
-}

--- a/src/app/index.ts
+++ b/src/app/index.ts
@@ -96,18 +96,38 @@ async function init(config?: IAppConfig) {
                   .then((res) => ({ stats: res }))
             : import('virtual:stats'));
 
-        console.log(stats);
-
         const updateSplashData = (data: TAppImportMap) => {
-            /** @todo implement with stats data */
             function updateSplashWithStats() {
-                const total =
-                    1 + Object.keys(data.assets).length + Object.keys(data.components).length;
-                const items =
-                    (data.lang !== undefined ? 1 : 0) +
-                    Object.values(data.assets).filter((flag) => flag).length +
-                    Object.values(data.components).filter((flag) => flag).length;
-                updateSplash((items / total) * 100);
+                function getAssetsSize(all = true): number {
+                    return Object.entries(data.assets)
+                        .filter(([, val]) => all || val)
+                        .map(
+                            ([key]) =>
+                                stats.assets[assetManifest[key].path.split('assets/').slice(-1)[0]],
+                        )
+                        .reduce((a, b) => a + b, 0);
+                }
+
+                function getComponentsSize(all = true): number {
+                    return Object.entries(data.components)
+                        .filter(([, val]) => all || val)
+                        .map(([key]) => stats.modules[key])
+                        .reduce((a, b) => a + b, 0);
+                }
+
+                const allAssetsSize = getAssetsSize();
+                const loadedAssetsSize = getAssetsSize(false);
+
+                const allComponentsSize = getComponentsSize();
+                const loadedComponentsSize = getComponentsSize(false);
+
+                const allLangSize = stats.i18n.en as number;
+                const loadedLangSize = data.lang ? (stats.i18n.en as number) : 0;
+
+                const totalSize = allAssetsSize + allComponentsSize + allLangSize;
+                const loadedSize = loadedAssetsSize + loadedComponentsSize + loadedLangSize;
+
+                updateSplash((loadedSize / totalSize) * 100);
             }
 
             if (import.meta.env.DEV && stats === undefined) {

--- a/src/app/index.ts
+++ b/src/app/index.ts
@@ -11,7 +11,7 @@ import type { TI18nLang } from '@/@types/core/i18n';
 // -------------------------------------------------------------------------------------------------
 
 const _importMap: TAppImportMap = {
-    lang: undefined,
+    lang: {},
     assets: {},
     components: {},
 };
@@ -27,7 +27,7 @@ function _updateImportMap(item: 'assets', subitem: string): TAppImportMap;
 function _updateImportMap(item: 'components', subitem: TComponentId): TAppImportMap;
 function _updateImportMap(item: keyof TAppImportMap, subitem?: string): TAppImportMap {
     if (item === 'lang') {
-        _importMap.lang = subitem as TI18nLang;
+        _importMap.lang[subitem as TI18nLang] = true;
     } else if (item === 'assets') {
         const assets = { ..._importMap.assets };
         assets[subitem as string] = true;
@@ -121,8 +121,9 @@ async function init(config?: IAppConfig) {
                 const allComponentsSize = getComponentsSize();
                 const loadedComponentsSize = getComponentsSize(false);
 
-                const allLangSize = stats.i18n.en as number;
-                const loadedLangSize = data.lang ? (stats.i18n.en as number) : 0;
+                const langName = Object.keys(data.lang)[0] as TI18nLang;
+                const allLangSize = stats.i18n[langName];
+                const loadedLangSize = data.lang[langName] ? allLangSize : 0;
 
                 const totalSize = allAssetsSize + allComponentsSize + allLangSize;
                 const loadedSize = loadedAssetsSize + loadedComponentsSize + loadedLangSize;
@@ -164,6 +165,7 @@ async function init(config?: IAppConfig) {
             }
         })();
 
+        _importMap.lang = Object.fromEntries([[importItemLang, false]]);
         _importMap.assets = Object.fromEntries(importListAssets.map((assetId) => [assetId, false]));
         _importMap.components = Object.fromEntries(
             importListComponents.map((componentId) => [componentId, false]),

--- a/src/app/index.ts
+++ b/src/app/index.ts
@@ -90,15 +90,21 @@ async function init(config?: IAppConfig) {
     {
         const { updateSplash } = await import('@/core/view');
 
-        const { stats } = await (import.meta.env.PROD
-            ? fetch('stats.json')
-                  .then((res) => res.json())
-                  .then((res) => ({ stats: res }))
-            : import('virtual:stats'));
+        const stats = import.meta.env.PROD
+            ? await fetch('stats.json').then((res) => res.json())
+            : undefined;
 
         const updateSplashData = (data: TAppImportMap) => {
-            function updateSplashWithStats() {
-                function getAssetsSize(all = true): number {
+            if (import.meta.env.DEV) {
+                const total =
+                    1 + Object.keys(data.assets).length + Object.keys(data.components).length;
+                const items =
+                    (data.lang !== undefined ? 1 : 0) +
+                    Object.values(data.assets).filter((flag) => flag).length +
+                    Object.values(data.components).filter((flag) => flag).length;
+                updateSplash((items / total) * 100);
+            } else {
+                const getAssetsSize = (all = true): number => {
                     return Object.entries(data.assets)
                         .filter(([, val]) => all || val)
                         .map(
@@ -106,14 +112,14 @@ async function init(config?: IAppConfig) {
                                 stats.assets[assetManifest[key].path.split('assets/').slice(-1)[0]],
                         )
                         .reduce((a, b) => a + b, 0);
-                }
+                };
 
-                function getComponentsSize(all = true): number {
+                const getComponentsSize = (all = true): number => {
                     return Object.entries(data.components)
                         .filter(([, val]) => all || val)
                         .map(([key]) => stats.modules[key])
                         .reduce((a, b) => a + b, 0);
-                }
+                };
 
                 const allAssetsSize = getAssetsSize();
                 const loadedAssetsSize = getAssetsSize(false);
@@ -129,18 +135,6 @@ async function init(config?: IAppConfig) {
                 const loadedSize = loadedAssetsSize + loadedComponentsSize + loadedLangSize;
 
                 updateSplash((loadedSize / totalSize) * 100);
-            }
-
-            if (import.meta.env.DEV && stats === undefined) {
-                const total =
-                    1 + Object.keys(data.assets).length + Object.keys(data.components).length;
-                const items =
-                    (data.lang !== undefined ? 1 : 0) +
-                    Object.values(data.assets).filter((flag) => flag).length +
-                    Object.values(data.components).filter((flag) => flag).length;
-                updateSplash((items / total) * 100);
-            } else {
-                updateSplashWithStats();
             }
         };
 

--- a/tools/vite.config.ts
+++ b/tools/vite.config.ts
@@ -63,35 +63,6 @@ export default defineConfig({
             gzipSize: true,
             template: 'raw-data',
         }),
-
-        // only for use in DEV mode
-        {
-            name: 'stats-plugin',
-            resolveId(id) {
-                if (id === 'virtual:stats') {
-                    return '\0' + 'virtual:stats';
-                }
-            },
-            load(id) {
-                if (id === '\0' + 'virtual:stats') {
-                    let stats:
-                        | {
-                              i18n: Record<string, number>;
-                              assets: Record<string, number>;
-                              modules: Record<string, number>;
-                          }
-                        | undefined;
-                    try {
-                        // if build was run before, this will be present
-                        // eslint-disable-next-line
-                        stats = require('../dist/stats.json');
-                    } catch (e) {
-                        // do nothing
-                    }
-                    return `export const stats = ${JSON.stringify(stats)}`;
-                }
-            },
-        },
     ],
 
     resolve: {

